ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -108,10 +112,57 @@ prop_Mempool_InvalidTxsNeverAdded bc txs =
     genTxIsInvalid (TestGenTx (InvalidTestTx _)) = True
     genTxIsInvalid _                             = False
 
+-- | Test that all valid transactions added to a 'Mempool' via 'addTxs' are
+-- appropriately represented in the trace of events.
+prop_Mempool_TraceValidTxs :: Chain TestBlock
+                           -> [TestTx]
+                           -> Property
+prop_Mempool_TraceValidTxs bc txs =
+    testAddTxsWithTrace bc txs traceProp
+  where
+    traceProp :: [GenTx TestBlock] -> [TraceEventMempool TestBlock] -> Property
+    traceProp genTxs es = case (filter isAddTxsEvent es) of
```

Comment:
Do we actually need the pattern-match here?
```haskell
sortGenTxs (filter genTxIsValid genTxs) === sortGenTxs addedTxs
```
works for both an empty and a non-empty list of events. (same for the next test)

Reply:
:man_facepalming: